pkg/cache/cache.go

```diff
@@ -11,19 +12,54 @@ const (
 	Megabyte = 1024 * 1024
 )
 
+// instance of default filesystem
+var defaultFS = &defaultFileSystem{}
```

Comment:
I understand why you have this here again. But it doesn't look right to have the same code repeated in almost every package. May be a better approach is to define both the interface for Os and it's default implementation under `pkg`  directly.

Then it can be reused by multiple pkg directly

```yaml
- pkg:
   -  os.go
```

I think the interface should better have an different name that is more generic say `SystemOS` since most of the functions are just wrapper around the `os` standard package.

And `DefaultSystemOS` then can be the pointer to the default implementation

Reply:
You can also provide a `os_test.go` file under pkg that has a public implementation of the mock so you can easily reuse it in all your test code

Reply:
We can has another interface `SystemCall` also for the syscall and so on. What do you think?